src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.GenerateAssemblyInfo.targets

```diff
@@ -84,7 +80,33 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_Parameter1>$(NeutralLanguage)</_Parameter1>
       </AssemblyAttribute>
     </ItemGroup>
+  </Target>
+
+  <!-- 
+    To allow version changes to be respected on incremental builds (e.g. through CLI parameters),
+    create a hash of all assembly attributes so that the output file name will change with the calculated
+    assembly attribute values and msbuild will then execute CoreGenerateAssembly to generate the new file.
+  -->
+  <Target Name="_CalculateGeneratedAssemblyInfoFileName" DependsOnTargets="_CalculateAssemblyAttributes">
+    <Hash ItemsToHash="@(AssemblyAttribute->'%(Identity)%(_Parameter1)')">
+      <Output TaskParameter="HashResult" PropertyName="_AssemblyAttributeContentHash" />
+    </Hash>
 
+    <PropertyGroup>
+      <GeneratedAssemblyInfoFile Condition=" '$(_AssemblyAttributeContentHash)' != '' ">
+        $(IntermediateOutputPath)$(MSBuildProjectName).AssemblyInfo.$(_AssemblyAttributeContentHash.SubString(0,6))$(DefaultLanguageSourceExtension)
```

Comment:
No, I was trying to figure out how much is actually needed to have a high probability of changing but not get too long and ugly. Since this is autogenerated and purely aesthetic (like short git commit hashes), i might was well use the full string.
Any preference?